### DISCUSSION TITLE
How to resend Verification

### DISCUSSION THREAD

**Discussion by Big-Brother-Man** [ory/.github#47] [Q&A] [answered]
How do I resend a verification email, even though that verification has already been sent a few hours ago and might not have expired (user didn't see it)

The admin api allows for recovery code but I don't need to recover the user just resend the verification email (with a new code if possible)

**Comment by vinckr** [comment, marked as answer]
Hello @Big-Brother-Man 

See this comment: https://github.com/ory/kratos/discussions/3036#discussioncomment-4820275

**Reply by Big-Brother-Man** [reply]
hey @vinckr many thanks or the kind reply. I had time to play around with the verify update path. I think I can articulate my problem a little better now. Here is another crack at it.

Tl;dr

I don't have the verification flow therefore I have to take a full round trip to get it.

When a user is registered, they are sent a email containing a code.

For my particular scenario we removed the link that contains the flow id.

We are left with the code.

However, the code does not contain the flow id for the given verification.

This leaves me with 1 options (please do add if you think there is other options available to me).

1. Go through the verification path (create and update) fully and use the response to get the verification flow id + code

Pass the verification flow from step 1 to a new page with the otp (one time password) input and await the users code.

Since Ory sent an email to the user, is there no other way to capture verification flow in the creation of the user account? This would save me from the round trip.

If not I have to create a middle page to create the verification flow + submit it then another page to receive it (since I have no idea what the flow is on user account creation nor in the email since we only show the code, this would also be an issue if a user just copied the code since by default I would not know the flow id). 

Another pain point is if I do a full round trip (create + update then pass in the response verification flow to a new page), user receives 2 emails not 1. 

In essence, how can I get the verification flow id for a new account without starting from creation + update?

I hope that made sense. If not please let me know.

Note:

Not sure how important this is but user can sign up while not verified but we don't continue the process till they are verified.

**Reply by Big-Brother-Man** [reply]
Here is a small diagram (used paint sorrrry!)

![image](https://user-images.githubusercontent.com/120871736/215909653-cfde1906-1198-401b-9ef7-baf91e294de6.png)

Hope that makes sense but this solution would still have sent 2 emails to the user :/

**Reply by Big-Brother-Man** [reply]
hey @vinckr thank you for your answer thinking about this further you have answered the question. I think I need to to open another post to address this other issue I am facing.

Thank you for your help.

**Reply by vinckr** [reply]
Sure no problem @Big-Brother-Man!
